CHI_RESOURCES/configure_dependencies.py

```diff
@@ -0,0 +1,506 @@
+    print("Downloading Random123-1.13.2 to \"" + os.getcwd() + "\"")
+    success,err = ExecSub("wget " + r123_url,log_file)
+
+  #Check if it is installed already
+  installed = os.path.exists(install_dir + "/RANDOM123/Random123-1.13.2/include")
+  if (not installed):
+    print("Configuring Random123-1.13.2 to \"" + os.getcwd() + "\"")
+    success,err = ExecSub("tar -zxf Random123-1.13.2.tar.gz",log_file)
+  else:
+    print("RANDOM123 already installed")
+
+  os.chdir(install_dir)
+
+####################################### Install triangle
+tri_url = "http://www.netlib.org/voronoi/triangle.zip"
+def InstallTriangle():
+  if (not os.path.exists(install_dir + "/TRIANGLE")):
+    os.mkdir("TRIANGLE")
+  os.chdir("TRIANGLE")
+
+  if (not os.path.exists(install_dir + "/TRIANGLE/triangle.zip")):
+    print("Downloading Triangle to \"" + os.getcwd() + "\"")
+    success,err = ExecSub("wget " + tri_url,log_file)
+
+  #Check if it is installed already
+  installed = os.path.exists(install_dir + "/TRIANGLE/triangle.o")
+  if (not installed):
+    print("Configuring Triangle to \"" + os.getcwd() + "\"")
+    success,err = ExecSub("unzip triangle.zip",log_file)
+
+    os_tag = "LINUX"
+    if ("Darwin" in os.uname() ):
+      os_tag = "APPLE"
+
+    env_vars=os.environ.copy()
+    command = "gcc -O -D" + os_tag + " -I/usr/X11R6/include" + \
+              " -L/usr/X11R6/lib -o ./triangle " + \
+              " ./triangle.c -lm"
+    success,err = ExecSub(command,log_file,env_vars)
+    print(command,err)
+
+    command = "gcc -O -D" + os_tag + " -I/usr/X11R6/include " + \
+              "-L/usr/X11R6/lib -DTRILIBRARY -c -o " + \
+              "./triangle.o ./triangle.c"
+    success,err = ExecSub(command,log_file,env_vars)
+    print(command,err)
+  else:
+    print("Triangle already installed")
+
+  os.chdir(install_dir)
+
+####################################### Install readline
+readline_url = "ftp://ftp.gnu.org/gnu/readline/readline-8.0.tar.gz"
+def InstallReadline():
+  if (not os.path.exists(install_dir + "/READLINE")):
+    os.mkdir("READLINE")
+  os.chdir("READLINE")
+
+  if (not os.path.exists(install_dir + "/READLINE/readline-8.0.tar.gz")):
+    print("Downloading Readline 8.0 to \"" + os.getcwd() + "\"")
+    success,err = ExecSub("wget " + readline_url,log_file)
+
+  #Check if it is installed already
+  installed = os.path.exists(install_dir + "/READLINE/readline-8.0"
+                                           "/build/lib/libreadline.a")
+  if (not installed):
+    print("Configuring Readline 8.0 to \"" + os.getcwd() + "\"")
+    success,err = ExecSub("tar -zxf readline-8.0.tar.gz",log_file)
+
+    os.chdir("readline-8.0")
+    env_vars=os.environ.copy()
+    command = "./configure --prefix=" + \
+              install_dir + \
+              "/READLINE/readline-8.0/build"
+    success,err = ExecSub(command,log_file,env_vars)
+    if (not success):
+      print(command,err)
+
+    command = "make -j8"
```

Comment:
I think it might be useful (here and at other instances of `make -j8`) to allow an additional optional argument to specify the number of parallel makes (you can keep the default 8 if not specified). For people with older machines (like me :-)), overloading the machine might really slow down the whole build process of the larger dependencies (e.g. VTK), and on the other hand, I can imagine people wanting to use more

Reply:
I have another update planned and it is something I also identified as a feature. Will include it there